src/test/versionbits_tests.cpp

```diff
@@ -253,6 +252,28 @@ BOOST_AUTO_TEST_CASE(versionbits_test)
                            .Mine(6000, TestTime(20000), 0).TestFailed().TestStateSinceHeight(6000)
                            .Mine(7000, TestTime(20000), 0x100).TestFailed().TestStateSinceHeight(6000)
                            .Mine(24000, TestTime(20000), 0x100).TestFailed().TestStateSinceHeight(6000) // stay in FAILED no matter how much we signal
+
+        // DEFINED -> STARTED -> LOCKEDIN -> ACTIVE (mandatory lockin)
```

Comment:
Please review this test thoroughly to ensure you understand and agree with the state transitions. See comment in src/versionbits.cpp for an explanation of the versionbits changes we are testing here.